src/cupbearer/tasks/_config.py

```diff
@@ -79,12 +79,37 @@ def _get_anomalous_test_data(self) -> DatasetConfig:
     def _init_model(self):
         pass
 
-    def build_train_data(self) -> Dataset:
+    def build_train_data(self, normal_weight: float = 1.0) -> Dataset:
         if not self._train_data:
             self._init_train_data()
             assert self._train_data is not None, "init_train_data must set _train_data"
             self._train_data.max_size = self.max_train_size
-        return self._train_data.build()
+
+        if normal_weight == 1.0:
+            return self._train_data.build()
+        else:
+            # SpectralDetector uses poisoned data for training
+            anomalous = self._get_anomalous_test_data()
+
+            # As TestDataMix adds a label for poisoned or not, we remove this here
+            class RemoveMixLabelDataset(Dataset):
```

Comment:
I wouldn't define this inline (e.g. maybe someone wants to check `isinstance(..., RemoveMixLabelDataset)` for some reason, also not sure whether there could be any issues with pickling this dataset if it's defined inline). Probably won't end up mattering though

Reply:
Sure,
 - [ ] I'll move it outside